### PR TITLE
[language][vm] split Container::General/Resource into Container::Locals/VecR/VecC/StructR/StructC

### PR DESCRIPTION
**Pull request by vgao1996**
## Summary
This replaces `Container::General/Resource` with `Container::Locals/VecR/VecC/StructR/StructC` to allow clearer separation of roles. 

## Test Plan
cargo test

### CODE REVIEW

File: language/move-vm/types/src/values/values_impl.rs [hunk 1]
```diff
@@ -67,13 +67,16 @@ enum ValueImpl {
 /// making it possible to be shared by references.
 #[derive(Debug)]
 enum Container {
```
[the remark this answers is not in the document]
Comment (vgao1996) [reply language/move-vm/types/src/values/values_impl.rs:69]: Oh I thought we talked about this... that's coming in the next PR!

File: language/move-vm/types/src/values/values_impl.rs [hunk 2]
```diff
@@ -562,55 +596,85 @@ impl IndexedRef {
...
-            | (General(v1), Resource(v2))
-            | (Resource(v1), General(v2))
-            | (General(v1), General(v2)) => v1[self.idx].equals(&v2[other.idx])?,
-
-            (U8(v1), U8(v2)) => v1[self.idx] == v2[other.idx],
-            (U64(v1), U64(v2)) => v1[self.idx] == v2[other.idx],
-            (U128(v1), U128(v2)) => v1[self.idx] == v2[other.idx],
-            (Bool(v1), Bool(v2)) => v1[self.idx] == v2[other.idx],
-            (Address(v1), Address(v2)) => v1[self.idx] == v2[other.idx],
+            // VecC <=> VecR impossible
```
[the remark this answers is not in the document]
Comment (vgao1996) [reply language/move-vm/types/src/values/values_impl.rs:599]: That's a good point. Let me try to clean this up a little bit.

**Comment by vgao1996** [comment]
@dariorussi No `is_valid_arg` is not updated in this PR but I'll be working on it soon.

File: language/move-vm/types/src/values/values_impl.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dariorussi) [reply language/move-vm/types/src/values/values_impl.rs:69]: got it! thanks

File: language/move-vm/types/src/values/values_impl.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dariorussi) [reply language/move-vm/types/src/values/values_impl.rs:599]: let's do this possible cleanup in a follow up so we can move forward and it helps with review

**Pull request closed without merging**